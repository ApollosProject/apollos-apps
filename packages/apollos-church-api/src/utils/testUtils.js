import getContext from 'apollos-church-api/src/getContext';
import getDataSources from 'apollos-church-api/src/getDataSources';
import { KeyValueCache } from 'apollo-server-caching';

export function getTestContext(req) {
  const context = getContext(req);

  const dataSources = getDataSources();
  // Apollo Server does this internally.
  Object.values(dataSources).forEach((dataSource) => {
    if (dataSource.initialize) {
      dataSource.initialize({ context, cache: KeyValueCache });
    }
  });
  context.dataSources = dataSources;
  return context;
}

export const buildGetMock = (response, dataSource) => {
  const get = jest.fn();
  if (Array.isArray(response) && Array.isArray(response[0])) {
    response.forEach((responseVal) => {
      get.mockReturnValueOnce(
        new Promise((resolve) => resolve(dataSource.normalize(responseVal)))
      );
    });
  }
  get.mockReturnValue(
    new Promise((resolve) => resolve(dataSource.normalize(response)))
  );
  return get;
};
