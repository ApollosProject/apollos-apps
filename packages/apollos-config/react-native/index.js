import NativeConfig from 'react-native-config';

class Config {
  config = {
    ...NativeConfig,
  };

  constructor() {
    this.attachConfigToClass(this.config);
  }

  loadJs(data = {}) {
    Object.keys(data).forEach((key) => {
      if (this.config[key] && typeof this.config[key] === 'object') {
        // Deep merge defaults if they exist.
        this.config[key] = Object.assign(this.config[key], data[key]);
      } else {
        // Otherwise attach results to config.
        this.config[key] = data[key];
      }
    });
    this.attachConfigToClass(this.config);
  }

  // Allows you to do Config.PROPERTY
  attachConfigToClass(config) {
    Object.keys(config).forEach((key) => {
      Object.defineProperty(this, key, {
        enumerable: true,
        configurable: true,
        writable: true,
        value: config[key],
      });
    });
  }
}

const config = new Config();

export default config;
