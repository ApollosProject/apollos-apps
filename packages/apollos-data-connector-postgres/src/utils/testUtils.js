/* eslint-disable import/prefer-default-export */
import createMigrationRunner from '../postgres/performMigrations';
import { sequelize } from '../postgres';

export const setupPostgresTestEnv = async (models = [], context) => {
  if (context.church.slug !== 'global') {
    await setupPostgresTestEnv(models, { church: { slug: 'global' } });
  }

  try {
    // If we don't have a set of migrations.
    // Hacky way of saying - have the migrations been run?
    await sequelize
      .getQueryInterface()
      .sequelize.query('select * from "SequelizeMeta"');
  } catch (e) {
    const migrationRunner = await createMigrationRunner({
      logger: null,
    });
    await migrationRunner.up();
  }

  await Promise.all(
    models.map((m) => m.models.createModel && m.models.createModel(context))
  );

  await Promise.all(
    models.map((m) => m.models.setupModel && m.models.setupModel(context))
  );
};
