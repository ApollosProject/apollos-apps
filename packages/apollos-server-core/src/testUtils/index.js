/* eslint-disable import/prefer-default-export */
import { makeExecutableSchema } from 'apollo-server';
import { testSchema } from '@apollosproject/data-schema';
import { KeyValueCache } from 'apollo-server-caching';
import { createApolloServerConfig } from '..';

export const createTestHelpers = (models) => {
  const serverConfig = createApolloServerConfig(models);
  const getContext = (req) => {
    const testContext = serverConfig.context(req);
    const testDataSources = serverConfig.dataSources();
    // Apollo Server does this internally.
    Object.values(testDataSources).forEach((dataSource) => {
      if (dataSource.initialize) {
        dataSource.initialize({ context: testContext, cache: KeyValueCache });
      }
    });
    testContext.dataSources = testDataSources;
    return testContext;
  };
  const getSchema = (schemas = []) =>
    makeExecutableSchema({
      typeDefs: [...serverConfig.schema, ...schemas, testSchema],
      resolvers: serverConfig.resolvers,
      resolverValidationOptions: {
        requireResolversForResolveType: false,
      },
    });

  return { getSchema, getContext };
};
