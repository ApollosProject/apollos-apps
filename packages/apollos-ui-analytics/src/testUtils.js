import React from 'react';

import { Providers } from '@apollosproject/ui-kit';
import { ApolloProvider } from 'react-apollo';
import { ApolloClient } from 'apollo-client';
import { InMemoryCache } from 'apollo-cache-inmemory';
import { createHttpLink } from 'apollo-link-http';
import AnalyticsProvider from './Provider';

const cache = new InMemoryCache();

export const trackMock = jest.fn();

const client = new ApolloClient({
  link: createHttpLink({ fetch: jest.fn() }),
  cache,
  resolvers: {},
});

export const wait = (time) =>
  new Promise((resolve) => setTimeout(resolve, time));

// eslint-disable-next-line
export const ProviderStack = ({ children }) => (
  <ApolloProvider client={client}>
    <Providers>
      <AnalyticsProvider trackFunctions={[trackMock]}>
        {children}
      </AnalyticsProvider>
    </Providers>
  </ApolloProvider>
);
