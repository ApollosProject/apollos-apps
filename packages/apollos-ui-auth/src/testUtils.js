/* eslint-disable import/prefer-default-export */
import React from 'react';
import { Providers as UIProviders } from '@apollosproject/ui-kit';
import { MockedProvider } from 'react-apollo/test-utils';

// eslint-disable-next-line
export const Providers = ({ children, ...props }) => (
  <UIProviders {...props}>
    <MockedProvider {...props}>{children}</MockedProvider>
  </UIProviders>
);
