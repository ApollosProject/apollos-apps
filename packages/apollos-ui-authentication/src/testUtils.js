import React from 'react';
import PropTypes from 'prop-types';
import { Providers as UIProviders } from '@apollosproject/ui-kit';
import { MockedProvider } from '@apollo/client/testing';

export const Providers = ({ children, ...props }) => (
  <UIProviders {...props}>
    <MockedProvider {...props}>{children}</MockedProvider>
  </UIProviders>
);

Providers.propTypes = {
  children: PropTypes.node,
};
