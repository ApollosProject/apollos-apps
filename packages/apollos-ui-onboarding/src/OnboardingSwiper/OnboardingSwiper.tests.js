import React from 'react';
import { TouchableOpacity, Text } from 'react-native';
import renderer from 'react-test-renderer';
import { Providers } from '../testUtils';
import OnboardingSwiper from '.';

describe('The onboarding swiper', () => {
  it('should render with a fragment as a child', () => {
    const tree = renderer.create(
      <Providers>
        <OnboardingSwiper>
          {({ swipeForward, scrollBy }) => (
            <>
              <TouchableOpacity onPress={swipeForward}>
                <Text>1</Text>
              </TouchableOpacity>
              <TouchableOpacity onPress={swipeForward}>
                <Text>2</Text>
              </TouchableOpacity>
              <TouchableOpacity key="3" onPress={() => scrollBy(-1)}>
                <Text>3</Text>
              </TouchableOpacity>
            </>
          )}
        </OnboardingSwiper>
      </Providers>
    );

    expect(tree).toMatchSnapshot();
  });
  it('should render with an array as a child', () => {
    const tree = renderer.create(
      <Providers>
        <OnboardingSwiper>
          {({ swipeForward, scrollBy }) => [
            <TouchableOpacity key="1" onPress={swipeForward}>
              <Text>1</Text>
            </TouchableOpacity>,
            <TouchableOpacity key="2" onPress={swipeForward}>
              <Text>2</Text>
            </TouchableOpacity>,
            <TouchableOpacity key="3" onPress={() => scrollBy(-1)}>
              <Text>3</Text>
            </TouchableOpacity>,
          ]}
        </OnboardingSwiper>
      </Providers>
    );

    expect(tree).toMatchSnapshot();
  });
});
