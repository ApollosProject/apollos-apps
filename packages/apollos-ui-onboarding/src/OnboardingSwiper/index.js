import React, { Component } from 'react';
import { View } from 'react-native';
import PropTypes from 'prop-types';
import * as ReactIs from 'react-is';

import Swiper from 'react-native-swiper';
import { styled } from '@apollosproject/ui-kit';

import { SafeAreaView } from 'react-navigation';

const dotStyles = ({ theme }) => ({
  width: theme.sizing.baseUnit / 2,
  height: theme.sizing.baseUnit / 2,
  borderRadius: theme.sizing.baseUnit / 4,
  margin: theme.sizing.baseUnit / 4,
});

const forceInset = {
  bottom: 'always',
};

const PaginationDot = styled(
  ({ theme }) => ({
    backgroundColor: theme.colors.background.inactive,
    ...dotStyles({ theme }),
  }),
  'ui-onboarding.OnboardingSwiper.PaginationDot'
)(View);

const PaginationDotActive = styled(
  ({ theme }) => ({
    backgroundColor: theme.colors.action.primary,
    ...dotStyles({ theme }),
  }),
  'ui-onboarding.OnboardingSwiper.PaginationDotActive'
)(View);

class OnboardingSwiper extends Component {
  static navigationOptions = () => ({
    title: 'Onboarding',
    header: null,
    gesturesEnabled: false,
  });

  static propTypes = {
    children: PropTypes.func.isRequired,
  };

  swiper = null;

  // Creates ref to Swiper to be passed as a prop to children.
  setSwiperRef = (r) => {
    this.swiper = r;

    return this.swiper;
  };

  // Advance swiper 1 slide. See Swiper documentation for scrollBy details. https://github.com/leecade/react-native-swiper#methods
  swipeForward = () => this.swiper.scrollBy(1);

  scrollBy = (...args) => this.swiper.scrollBy(...args);

  render() {
    const children = this.props.children({
      swipeForward: this.swipeForward,
      scrollBy: this.scrollBy,
    });

    // The user can pass either a set of components in a fragment, or an array of components.
    // If the uses a fragment, we need to remove the fragment and render the components directly.
    // Otherwise the slider will treat the fragment as a single slide.
    let slides = children;
    if (ReactIs.typeOf(children) === ReactIs.Fragment) {
      slides = children.props.children;
    }

    return (
      <Swiper
        loadMinimal
        loop={false}
        /* Disables swipe gestures. We currently we dont display a back button so this is our
         * only back navigation option. */
        // scrollEnabled={false}
        showsButtons={false}
        ref={this.setSwiperRef}
        renderPagination={this.renderPagination}
        activeDot={
          <SafeAreaView forceInset={forceInset}>
            <PaginationDotActive />
          </SafeAreaView>
        }
        dot={
          <SafeAreaView forceInset={forceInset}>
            <PaginationDot forceInset={forceInset} />
          </SafeAreaView>
        }
        {...this.props}
      >
        {slides}
      </Swiper>
    );
  }
}

export default OnboardingSwiper;
