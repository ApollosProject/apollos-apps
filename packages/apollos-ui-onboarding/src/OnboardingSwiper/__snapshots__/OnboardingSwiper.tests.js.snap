// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The onboarding swiper should render with a fragment as a child 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
        "position": "relative",
      },
      undefined,
    ]
  }
>
  <RCTScrollView
    activeDot={
      <ForwardRef(_SafeAreaView)
        forceInset={
          Object {
            "bottom": "always",
          }
        }
      >
        <mapProps(getContext(mapProps(withPropsOnChange(mapProps(View))))) />
      </ForwardRef(_SafeAreaView)>
    }
    automaticallyAdjustContentInsets={false}
    autoplay={false}
    autoplayDirection={true}
    autoplayTimeout={2.5}
    bounces={false}
    contentContainerStyle={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    contentOffset={
      Object {
        "x": 0,
      }
    }
    disableNextButton={false}
    dot={
      <ForwardRef(_SafeAreaView)
        forceInset={
          Object {
            "bottom": "always",
          }
        }
      >
        <mapProps(getContext(mapProps(withPropsOnChange(mapProps(View)))))
          forceInset={
            Object {
              "bottom": "always",
            }
          }
        />
      </ForwardRef(_SafeAreaView)>
    }
    horizontal={true}
    index={0}
    loadMinimal={true}
    loadMinimalSize={1}
    loop={false}
    onIndexChanged={[Function]}
    onMomentumScrollEnd={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    pagingEnabled={true}
    removeClippedSubviews={true}
    scrollsToTop={false}
    showsButtons={false}
    showsHorizontalScrollIndicator={false}
    showsPagination={true}
    showsVerticalScrollIndicator={false}
  >
    <View>
      <View
        style={
          Array [
            Object {
              "height": 1334,
              "width": 750,
            },
            Object {
              "backgroundColor": "transparent",
            },
          ]
        }
      >
        <View
          accessible={true}
          clickable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text>
            1
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "height": 1334,
              "width": 750,
            },
            Object {
              "backgroundColor": "transparent",
            },
          ]
        }
      >
        <View
          accessible={true}
          clickable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text>
            2
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "height": 1334,
            "justifyContent": "center",
            "width": 750,
          }
        }
      >
        <ActivityIndicator
          animating={true}
          color="#999999"
          hidesWhenStopped={true}
          size="small"
        />
      </View>
    </View>
  </RCTScrollView>
  <View
    pointerEvents="none"
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "bottom": 25,
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
        },
        undefined,
      ]
    }
  >
    <RCTSafeAreaView
      emulateUnlessSupported={true}
      forceInset={
        Object {
          "bottom": "always",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#00676D",
            "borderRadius": 4,
            "height": 8,
            "margin": 4,
            "width": 8,
          }
        }
      />
    </RCTSafeAreaView>
    <RCTSafeAreaView
      emulateUnlessSupported={true}
      forceInset={
        Object {
          "bottom": "always",
        }
      }
    >
      <View
        forceInset={
          Object {
            "bottom": "always",
          }
        }
        style={
          Object {
            "backgroundColor": "#A5A5A5",
            "borderRadius": 4,
            "height": 8,
            "margin": 4,
            "width": 8,
          }
        }
      />
    </RCTSafeAreaView>
    <RCTSafeAreaView
      emulateUnlessSupported={true}
      forceInset={
        Object {
          "bottom": "always",
        }
      }
    >
      <View
        forceInset={
          Object {
            "bottom": "always",
          }
        }
        style={
          Object {
            "backgroundColor": "#A5A5A5",
            "borderRadius": 4,
            "height": 8,
            "margin": 4,
            "width": 8,
          }
        }
      />
    </RCTSafeAreaView>
  </View>
</View>
`;

exports[`The onboarding swiper should render with an array as a child 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
        "position": "relative",
      },
      undefined,
    ]
  }
>
  <RCTScrollView
    activeDot={
      <ForwardRef(_SafeAreaView)
        forceInset={
          Object {
            "bottom": "always",
          }
        }
      >
        <mapProps(getContext(mapProps(withPropsOnChange(mapProps(View))))) />
      </ForwardRef(_SafeAreaView)>
    }
    automaticallyAdjustContentInsets={false}
    autoplay={false}
    autoplayDirection={true}
    autoplayTimeout={2.5}
    bounces={false}
    contentContainerStyle={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    contentOffset={
      Object {
        "x": 0,
      }
    }
    disableNextButton={false}
    dot={
      <ForwardRef(_SafeAreaView)
        forceInset={
          Object {
            "bottom": "always",
          }
        }
      >
        <mapProps(getContext(mapProps(withPropsOnChange(mapProps(View)))))
          forceInset={
            Object {
              "bottom": "always",
            }
          }
        />
      </ForwardRef(_SafeAreaView)>
    }
    horizontal={true}
    index={0}
    loadMinimal={true}
    loadMinimalSize={1}
    loop={false}
    onIndexChanged={[Function]}
    onMomentumScrollEnd={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    pagingEnabled={true}
    removeClippedSubviews={true}
    scrollsToTop={false}
    showsButtons={false}
    showsHorizontalScrollIndicator={false}
    showsPagination={true}
    showsVerticalScrollIndicator={false}
  >
    <View>
      <View
        style={
          Array [
            Object {
              "height": 1334,
              "width": 750,
            },
            Object {
              "backgroundColor": "transparent",
            },
          ]
        }
      >
        <View
          accessible={true}
          clickable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text>
            1
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "height": 1334,
              "width": 750,
            },
            Object {
              "backgroundColor": "transparent",
            },
          ]
        }
      >
        <View
          accessible={true}
          clickable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text>
            2
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "height": 1334,
            "justifyContent": "center",
            "width": 750,
          }
        }
      >
        <ActivityIndicator
          animating={true}
          color="#999999"
          hidesWhenStopped={true}
          size="small"
        />
      </View>
    </View>
  </RCTScrollView>
  <View
    pointerEvents="none"
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "bottom": 25,
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
        },
        undefined,
      ]
    }
  >
    <RCTSafeAreaView
      emulateUnlessSupported={true}
      forceInset={
        Object {
          "bottom": "always",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#00676D",
            "borderRadius": 4,
            "height": 8,
            "margin": 4,
            "width": 8,
          }
        }
      />
    </RCTSafeAreaView>
    <RCTSafeAreaView
      emulateUnlessSupported={true}
      forceInset={
        Object {
          "bottom": "always",
        }
      }
    >
      <View
        forceInset={
          Object {
            "bottom": "always",
          }
        }
        style={
          Object {
            "backgroundColor": "#A5A5A5",
            "borderRadius": 4,
            "height": 8,
            "margin": 4,
            "width": 8,
          }
        }
      />
    </RCTSafeAreaView>
    <RCTSafeAreaView
      emulateUnlessSupported={true}
      forceInset={
        Object {
          "bottom": "always",
        }
      }
    >
      <View
        forceInset={
          Object {
            "bottom": "always",
          }
        }
        style={
          Object {
            "backgroundColor": "#A5A5A5",
            "borderRadius": 4,
            "height": 8,
            "margin": 4,
            "width": 8,
          }
        }
      />
    </RCTSafeAreaView>
  </View>
</View>
`;
